lib/cisco_node_utils/portchannel_global.rb

```diff
@@ -0,0 +1,303 @@
+
+      create if instantiate
+    end
+
+    def self.globals
+      hash = {}
+      # if the key already exists, no need create instance
+      hash['default'] =
+          PortChannelGlobal.new('default', false) unless hash.key?('default')
+      hash
+    end
+
+    # create and destroy are not needed for this as
+    # portchannel global commands do not have any
+    # pre-requisites
+    def create
+    end
+
+    def destroy
+      @name = nil
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def hash_distribution
+      config_get('portchannel_global', 'hash_distribution')
+    end
+
+    def hash_distribution=(val)
+      config_set('portchannel_global',
+                 'hash_distribution', val)
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_hash_distribution
+      config_get_default('portchannel_global', 'hash_distribution')
+    end
+
+    def load_defer
+      config_get('portchannel_global', 'load_defer')
+    end
+
+    def load_defer=(val)
+      config_set('portchannel_global', 'load_defer', val)
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_load_defer
+      config_get_default('portchannel_global', 'load_defer')
+    end
+
+    def resilient
+      config_get('portchannel_global', 'resilient')
+    end
+
+    def resilient=(state)
+      fail TypeError unless state == true || state == false
+      no_cmd = (state ? '' : 'no')
+      config_set('portchannel_global', 'resilient', no_cmd)
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_resilient
+      config_get_default('portchannel_global', 'resilient')
+    end
+
+    # port-channel load-balance is a complicated command
+    # and this has so many forms in the same device and in
+    # different devices.
+    # For ex:
+    # port-channel load-balance src-dst ip rotate 4 concatenation symmetric
+    # port-channel load-balance resilient
+    # port-channel load-balance ethernet destination-mac CRC10c
+    # port-channel load-balance ethernet source-ip
+    # port-channel load-balance dst ip-l4port rotate 4 asymmetric
+    # port-channel load-balance dst ip-l4port-vlan module 9
+    # port-channel load-balance hash-modulo
+    # we need to eliminate all the clutter and get the correct
+    # line of config first and after that get index of each property
+    # and get the property value because some properties may or
+    # may not be present always.
+    # This method returns a list
+    # 1st item is bundle_select
+    # 2nd item is bundle_hash
+    # 3rd item is hash_poly
+    # 4th item is asymmetric
+    # 5th item is symmetry
+    # 6th item is rotate
+    # 7th item is concatenation
```

Comment:
Ouch. This hurts my brain. :-)

I would prefer to have you use a key-val hash instead of the positional list you have here. It's usually much easier to maintain and extend.

With a hash, your getters simply do:

```
def asymmetric
  port_channel_load_balance[:asymmetric]
end
```